Error while checking build/O2Physics/o2 for 0ffbdd37c05c99ff51326842f06d0c35db71e3b8 at 2026-04-23 23:40:
```
## sw/BUILD/O2Physics-latest/log
/sw/SOURCES/O2Physics/slc9_x86-64-slc9_x86-64/0/PWGDQ/Tasks/qaMatching.cxx:1679:44: error: unused parameter 'muonTracks' [-Werror=unused-parameter]
ninja: build stopped: subcommand failed.
```
Full log [here](https://ali-ci.cern.ch/alice-build-logs/AliceO2Group/O2Physics/15953/0ffbdd37c05c99ff51326842f06d0c35db71e3b8/build_O2Physics_o2/pretty.html).

--- a/PWGDQ/Tasks/qaMatching.cxx
+++ b/PWGDQ/Tasks/qaMatching.cxx
@@ -139,6 +139,7 @@ struct QaMatching {
   static constexpr int ExtrapolationMethodMftFirstPoint = 2;
   static constexpr int ExtrapolationMethodVertex = 3;
   static constexpr int ExtrapolationMethodMftDca = 4;
+  static constexpr int DecayRankingDirect = 2;
 
   struct MatchingCandidate {
     int64_t collisionId{-1};
@@ -1675,7 +1676,7 @@ struct QaMatching {
 
   template <class TMUON, class TMUONS, class TMFTS>
   MuonMatchType getMatchType(const TMUON& muonTrack,
-                             TMUONS const& muonTracks,
+                             TMUONS const& /*muonTracks*/,
                              TMFTS const& mftTracks,
                              const std::vector<std::pair<int64_t, int64_t>>& matchablePairs,
                              int ranking)
@@ -1685,18 +1686,18 @@ struct QaMatching {
 
     auto const& mchTrack = muonTrack.template matchMCHTrack_as<TMUONS>();
 
-    bool isPaired = isMatchableMch(mchTrack.globalIndex(), matchablePairs);
-    bool isMuon = IsMuon(muonTrack, muonTracks, mftTracks);
+    bool isPairable = isMatchableMch(mchTrack.globalIndex(), matchablePairs);
+    bool isTrueMatch = isTrueGlobalMatching(muonTrack, matchablePairs);
     int decayRanking = getDecayRanking(mchTrack, mftTracks);
 
     MuonMatchType result{kMatchTypeUndefined};
-    if (isPaired) {
-      if (isMuon) {
+    if (isPairable) {
+      if (isTrueMatch) {
         result = (ranking == 1) ? kMatchTypeTrueLeading : kMatchTypeTrueNonLeading;
       } else {
         result = (ranking == 1) ? kMatchTypeWrongLeading : kMatchTypeWrongNonLeading;
       }
-    } else if (decayRanking == 1) {
+    } else if (decayRanking == DecayRankingDirect) {
       result = (ranking == 1) ? kMatchTypeDecayLeading : kMatchTypeDecayNonLeading;
     } else {
       result = (ranking == 1) ? kMatchTypeFakeLeading : kMatchTypeFakeNonLeading;

